src/commands/app/upgrade.tsx

```diff
@@ -0,0 +1,170 @@
+import { ExecRenderBaseCommand } from "../../rendering/react/ExecRenderBaseCommand.js";
+import {
+  appInstallationArgs,
+  withAppInstallationId,
+} from "../../lib/app/flags.js";
+import { projectFlags } from "../../lib/project/flags.js";
+import { Flags, ux } from "@oclif/core";
+import React, { ReactNode } from "react";
+import { Text } from "ink";
+import { getAppFromUuid, getAppVersionFromUuid } from "../../lib/app/uuid.js";
+import {
+  getAvailableTargetAppVersionFromExternalVersion,
+  getLatestAvailableTargetAppVersionForAppVersionUpgradeCandidates,
+} from "../../lib/app/versions.js";
+import {
+  makeProcessRenderer,
+  ProcessFlags,
+  processFlags,
+} from "../../rendering/process/process_flags.js";
+import { Success } from "../../rendering/react/components/Success.js";
+
+export class UpgradeApp extends ExecRenderBaseCommand<typeof UpgradeApp, void> {
+  static description = "Upgrade target app installation to target version";
+  static args = {
+    ...appInstallationArgs,
+  };
+  static flags = {
+    "target-version": Flags.string({
+      description: "target version to upgrade target app to",
+      required: true,
+      default: "latest",
+    }),
+    "show-candidates": Flags.boolean({
+      description: "list version candidates the given app can be upgraded to",
+    }),
```

Comment:
Listing potential target versions and actually performing an update are quite different things; maybe extract this functionality into a different command? Like `mw app list-upgrade-candidates`?

EDIT: That command would also make sense to be implemented as a `ListBaseCommand` subclass, because I could imagine a user wanting JSON/YAML output for that command (which would not be easily possible with the current implementation).

Reply:
Valid point - This will be implemented.
I would argue though, that since most of the same data is required for the upgrade process anyway, so giving the user the option to check the candidates while already preparing the upgrade command might make for a smoother workflow - Albeit this means that either the flag or the extra command would be redundant. I don't see the harm though except some degree of bloating the cli?

Reply:
Or, alternatively, while thinking about it some more: If no `--target-version` was specified, give the user the option to select the desired target version interactively (if used within a TTY context) from the candidate list; this could be done using the `ProcessRenderer`s `addSelect` method:

https://github.com/mittwald/cli/blob/d8076f7877613b86aaebabeb8bf55654003245fd/src/rendering/process/process.tsx#L116-L119